client-app/src/pages/account/views/Home.vue

```diff
@@ -1,88 +1,67 @@
-                            <div class="col-md">
-                                <label for="phone">{{ $t('customer.login.phone') }}</label>
-                                <input
-                                    type="text"
-                                    class="form-control"
-                                    id="phone"
-                                    v-model="profile.phoneNumber"
-                                />
-                            </div>
-                        </div>
-                    </fieldset>
-                    <a class="btn btn--green" @click.prevent="update()"
-                        >Update User</a
-                    >
-                </div>
+      <!-- Tabs work not js only css -->
+      <div class="tabs">
+        <input class="tabs__input" type="radio" name="tabs" id="tab-1" checked />
+        <label for="tab-1" class="tabs__item">
+          User Details
+          <!-- <span class="mobile-name">User Details</span> -->
+        </label>
+        <div class="tabs__content">
+          <fieldset>
+            <div class="form-row form-group">
+              <div class="col-md">
+                <label for="login">{{ $t("customer.login.user_name") }}</label>
+                <input type="text" class="form-control" id="login" v-model="profile.userName" readonly />
+              </div>
+              <div class="col-md">
+                <label for="email">{{ $t("customer.login.email") }}</label>
+                <input type="email" class="form-control" id="email" v-model="profile.email" />
+              </div>
+              <div class="col-md">
+                <label for="phone">{{ $t("customer.login.phone") }}</label>
+                <input type="text" class="form-control" id="phone" v-model="profile.phoneNumber" />
+              </div>
             </div>
+          </fieldset>
+          <a class="btn btn--green" @click.prevent="update()">Update User</a>
         </div>
-    </main>
+      </div>
+    </div>
+  </main>
 </template>
 
-<script>
-import { mapGetters } from "vuex";
-import { FETCH_PROFILE, UPDATE_USER } from "../store/index";
- // Import component
-import Loading from 'vue-loading-overlay';
+<script lang="ts">
+import Vue from "vue";
+import Component from "vue-class-component";
+import { FETCH_PROFILE, UPDATE_USER } from "@account/store/definitions";
+import Loading from "vue-loading-overlay";
+import 'vue-loading-overlay/dist/vue-loading.css';
 
-export default {
-    name: "ProfileHome",
-    
-    mounted() {        
-        this.$store.dispatch(FETCH_PROFILE, "userName");
-    },
-    components: {
-            Loading
-    },
-    computed: {
-        ...mapGetters(["profile", "isLoading"])
-    },
-    methods: {
-        update() {
-            this.$store.dispatch(UPDATE_USER, this.profile);
-        }
-    }
-};
+@Component({
+  name: "ProfileHome",
+  components: {
+    Loading
+  }
+})
+export default class Home extends Vue {
+  get profile() {
```

Comment:
Is this syntax sugar won't work here? ` ...mapGetters(["profile", "isLoading"])`

Reply:
No and it shouldn't. You loose type checking when you use strings